src/cli/commands/deposit.ts

```diff
@@ -0,0 +1,79 @@
+import {CliCommand} from "./interface";
+import * as commander from "commander";
+import defaults from "../../eth1/defaults";
+import * as ethers from "ethers/ethers";
+import {Wallet} from "ethers/ethers";
+import logger from "../../logger";
+import {Eth1Wallet} from "../../eth1";
+import {CliError} from "../error";
+
+export class DepositCommand implements CliCommand {
+
+  public register(commander: commander.CommanderStatic): void {
+    commander
+      .command('deposit')
+      .description('Start private network with deposit contract and 10 accounts with balance')
+      .option("-k, --privateKey [privateKey]", 'Private key of account that will make deposit')
+      .option("-m, --mnemonic [mnemonic]", 'If mnemonic is submitted, first 10 accounts will make deposit')
+      .option("-n, --node [node]", 'Url of eth1 node', 'http://127.0.0.1:8545')
+      .option("-v, --value [value]", 'Amount of ether to deposit', "32")
+      .option("-c, --contract [contract]", 'Address of deposit contract', defaults.depositContract.address)
+      .action( async ({privateKey, mnemonic, node, value, contract}) => {
+        //library is not awaiting this method so don't allow error propagation (unhandled promise rejections
+        try {
+          await this.action(privateKey, mnemonic, node, value, contract);
+        } catch (e) {
+          logger.error(e.message);
+        }
+
+      });
+  }
+
+  public async action(privateKey: string, mnemonic: string, node: string, value: string, contract: string) {
```

Comment:
Here I would make the contract optional and allow a default to be passed? Otherwise it looks as though current behaviour forces the user to pass the address

Reply:
Hmm I can, although user should in --help see that.

![image](https://user-images.githubusercontent.com/8836210/56039374-7985b880-5d34-11e9-873c-be4cd76f6bc3.png)

Reply:
oh wicked thats awesome! I'm on mobile so haven't been able to actually run the code!

I'll get to that right now!